packages/js/product-editor/src/lock-unlock.ts

```diff
@@ -0,0 +1,10 @@
+/**
+ * External dependencies
+ */
+import { __dangerousOptInToUnstableAPIsOnlyForCoreModules } from '@wordpress/private-apis';
+
+export const { lock, unlock } =
+	__dangerousOptInToUnstableAPIsOnlyForCoreModules(
+		'I acknowledge private features are not for use in themes or plugins and doing so will break in the next version of WordPress.',
+		'@wordpress/edit-site'
+	);
```

Comment:
This only works within Gutenberg enabled, which is why I added the additional check in the `products.tsx` compoennt.